plugins/api/src/switch_user.cpp

```diff
@@ -16,108 +16,181 @@
+    auto check_input(const SwitchUserInput* _input) -> int
     {
-        using log_api = irods::experimental::log::api;
-
-        // Only administrators are allowed to invoke this API.
-        // We check the proxy user because it covers clients and server-to-server redirects.
-        // Remember, the client and proxy user information is identical until there's a redirect.
-        if (!irods::is_privileged_proxy(*_comm)) {
-            log_api::error(
-                "Proxy user [{}] does not have permission to switch client user.", _comm->proxyUser.userName);
-            return SYS_PROXYUSER_NO_PRIV;
-        }
-
         if (!_input) {
-            log_api::error("Invalid input argument: Received a null pointer");
+            log_api::error("Invalid input argument: received a null pointer");
             return SYS_INVALID_INPUT_PARAM;
         }
 
-        const auto* local_zone = getLocalZoneName();
-
-        if (!local_zone) {
-            log_api::error("Could not get name of local zone.");
-            return SYS_INTERNAL_ERR;
-        }
-
-        const auto* username = _input->username;
-        const auto* zone = _input->zone;
-        bool is_local_zone = false;
-
-        // Passing an empty string as the zone argument is equivalent to explicitly setting
-        // the zone parameter to the local zone.
-        if (const std::string_view z{_input->zone}; z.empty() || z == local_zone) {
-            zone = local_zone;
-            is_local_zone = true;
+        if (std::strnlen(_input->username) == 0) {
+            log_api::error("Invalid username argument: null or empty");
+            return SYS_INVALID_INPUT_PARAM;
         }
 
-        namespace adm = irods::experimental::administration;
-
-        // This call covers the existence check too.
-        const auto user_type = adm::server::type(*_comm, adm::user{username, zone});
-
-        if (!user_type) {
-            log_api::error("[{}#{}] is not a user in the local zone.", username, zone);
+        if (std::strlen(_input->zone) == 0) {
+            log_api::error("Invalid zone argument: null or empty");
             return SYS_INVALID_INPUT_PARAM;
         }
 
-        const auto* user_type_string = adm::to_c_str(*user_type);
-        auto& client = _comm->clientUser;
+        return 0;
+    } // check_input
 
-        // Update the user identity associated with the RsComm.
-        std::strncpy(client.userName, username, sizeof(UserInfo::userName));
-        std::strncpy(client.rodsZone, zone, sizeof(UserInfo::rodsZone));
-        std::strncpy(client.userType, user_type_string, sizeof(UserInfo::userType));
+    auto update_server_to_server_connections(RsComm& _comm) -> void
+    {
+        const auto switch_user_or_disconnect = [&client = _comm.clientUser](rodsServerHost& _host)
+        {
+            if (!_host.conn) {
+                log_api::warn("No connection to remote host."); // TODO Consider including the host's name.
```

Comment:
don't see a downside - could help the admin discover the misconfiguration more easily.

Reply:
Will include the host info.

Reply:
done